spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/datastore/DatastoreKeyIdConverter.java

```diff
@@ -66,6 +66,11 @@ public String toRequestId(Serializable serializable, Class<?> aClass) {
 	public boolean supports(Class<?> entityType) {
 		// This ID converter only covers the Datastore key type. Returning false here causes the
 		// default converter from Spring Data to be used.
-		return this.datastoreMappingContext.getPersistentEntity(entityType).getIdProperty().getType().equals(Key.class);
+		try {
```

Comment:
I think catching exceptions is expensive compared to just checking the values. Maybe it's preferable to just check the values directly?

```
result = this.datastoreMappingContext.getPersistentEntity(entityType)

if (result == null) {
  return false;
} else {
  return result.getIdProperty().getType().equals(Key.class);
}
```

Reply:
`getIdProperty()` can also return null :(

Reply:
Ahh ok.